interface/modules/custom_modules/oe-module-ehi-exporter/src/EhiExporter.php

```diff
@@ -0,0 +1,723 @@
+<?php
+
+/**
+ * Main class for EhiExporter for exporting data from the db
+ *
+ * @package   OpenEMR
+ * @link      http://www.open-emr.org
+ *
+ * @author    Stephen Nielson <snielson@discoverandchange.com
+ * @copyright Copyright (c) 2023 OpenEMR Foundation, Inc
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+namespace OpenEMR\Modules\EhiExporter;
+
+use OpenEMR\Common\Crypto\CryptoGen;
+use OpenEMR\Common\Database\QueryUtils;
+use OpenEMR\Common\Logging\SystemLogger;
+use OpenEMR\Common\Twig\TwigContainer;
+use OpenEMR\Common\Utils\FileUtils;
+use OpenEMR\Common\Uuid\UuidRegistry;
+use OpenEMR\FHIR\Export\ExportException;
+use OpenEMR\Modules\EhiExporter\Models\EhiExportJob;
+use OpenEMR\Modules\EhiExporter\Models\EhiExportJobTask;
+use OpenEMR\Modules\EhiExporter\Services\EhiExportJobService;
+use OpenEMR\Modules\EhiExporter\Services\EhiExportJobTaskResultService;
+use OpenEMR\Modules\EhiExporter\Services\EhiExportJobTaskService;
+use OpenEMR\Services\DocumentService;
+use OpenEMR\Services\ListService;
+use OpenEMR\Modules\EhiExporter\ExportState;
+use OpenEMR\Modules\EhiExporter\ExportTableDefinition;
+use OpenEMR\Modules\EhiExporter\ExportKeyDefinition;
+use Ramsey\Uuid\Rfc4122\UuidV4;
+use Twig\Environment;
+
+class EhiExporter
+{
+    const EHI_DOCUMENT_CATEGORY = "EHI Export Zip File";
+
+    /**
+     * The folder name that export documents are stored in.
+     */
+    const EHI_DOCUMENT_FOLDER = 'system-ehi-export';
+
+    const PARENT_FK_TABLES_TRAVERSAL = ['patient_data', 'insurance_data', 'eligibility_verification', 'form_vitals', 'lbt_data',  'lbf_data', 'patient_tracker', 'therapy_groups'];
+    const ZIP_MIME_TYPE = "application/zip";
+    const PATIENT_TASK_BATCH_FETCH_LIMIT = 5000;
+    const CYCLE_MAX_ITERATIONS_LIMIT = 500;
```

Comment:
I bumped up the limit to 1500.  I did remove some of the extra query passes that were happening on the form tables so this may not be needed, but its better having this be larger than not.

Reply:
i'll pull down and retest after you finish with updates

Reply:
Looks like this is working now.